package.json

```diff
@@ -38,6 +38,9 @@
     "turbo": "^1.10.13",
     "vitest": "1.6.0"
   },
+  "engines": {
+    "node": "^18.0.0 || ^20.0.0"
+  },
```

Comment:
What effect does the engine spec have on the root level?

Reply:
An install or build command with a Node version that isn't `18.x.x` or `20.x.x` will output a warning as below at the start of install/build. I think this is fine, and preferred to (say) hard-failing, though we could do that too to force a maintainer to have a known good version (see the PR description for more).

There should be no effect for those with a compatible Node version.

Note that I believe non-maintainers installing the AI SDK with a Node engine that doesn't fit the spec will also see the warning. Again, this seems ok to me. We don't block/disallow the install, but we let them know their mileage may vary.

```
 WARN  Unsupported engine: wanted: {"node":"^18.0.0 || ^20.0.0"} (current: {"node":"v23.0.0","pnpm":"9.12.2"})
```

Reply:
The top level seemed like the right place for the engine spec, but if you think it makes more sense we could do it in `packages/ai/package.json` instead. The `pnpm` version is at the topmost level already.